lib/puppet/provider/openldap_database/olc.rb

```diff
@@ -49,7 +49,7 @@ def self.instances
         when %r{^olcRelay: }
           relay = line.split[1]
         when %r{^olcReadOnly: }i
-          readonly = line.split[1]
+          readonly = line.split[1] == 'TRUE' ? :true : :false
```

Comment:
I think we can use native types here:
```suggestion
          readonly = line.split[1] == 'TRUE'
```

Reply:
Well, I'm a bit confused at this, to be quite honest; the `mirrormode` setting defined the `:true` and `:false` values, which *appears* to be an accepted way as it's used all throughout various modules, including modules by @puppetlabs. So I just tried to mirror (hah) that, and also pushed f2e809d since the symbol approach needed that.

I could just drop it and go down to ruby booleans, but then there'd be a certain inconsistency, and one that the overall codebase, including @voxpupuli, seems split on. Or have I overlooked a style guide somewhere that defines the way to go?

Reply:
this won't help you, but: This is an ongoing problem since types and provider exists and everytime I don't know how to handle it properly :(

Reply:
Deprecation of `:true` / `:false` in favor of `true` / `false` is quite vague for me. I feel like it was needed with older parser and Puppet 4 fixed that. As you say, there are inconsistencies in the various modules around. Some module even mix both ways: https://github.com/puppetlabs/puppetlabs-vcsrepo/blob/bcae373622364f3088fd6d0c89c4d499c05b3e5d/lib/puppet/type/vcsrepo.rb#L290-L321

This makes me think we can switch to pure `true` / `false` but if it cause trouble that's not a major issue.